### PR TITLE
Teensy driving bearing halo

### PR DESCRIPTION
**Pull request by cburnout**
# Description

This is a PR that should probably be at least 3. Sorry
This has a few things 

Mostly it builds our rover docker built up more. it lets us update our heading either through a mock which just increments heading at 10hz or it lets you connect through rosserial to a teensy that has our INS on it. I have a spare teensy and board so we have 2 that we can use for development

This also has a renaming of EventSystem to ARSISEventSystem because when we include the oculus build event system is something they use and removes the UserSetting directory so we dont have the cruft being added to the repo.

Fixes # (issue)

#58 

## Bug Fix or Feature

- [ ] Bug
- [x] Feature
- [ ] Other

# Testing Instructions

- [ ] Start GC+T services
- [ ] Add a user so there is a user with `user_id` `1`and username of `user1`
- [ ] Check that `rover.env` has `VISION_KIT_MODE=PROTOTYPE`
- [ ] run the `Rover` docker compose file
- [ ] Start the bearing halo unity scene
- [ ] See that the bearing halo slowly rotates

## Linting

- [ ] I've linted my code

### CODE REVIEW

File: ARSIS-Rover/teensy/src/main.cpp
Comment (Aetheris743) [inline ARSIS-Rover/teensy/src/main.cpp:1]: Sorry, this is a a little specific, but you might want to run an autoformat on this file. Spacing seems to be a little inconsistent. (Especially around parens)

File: ARSIS-Unity/Assets/ARSIS/Core/EventManager/BasesArsisEvent.cs
Comment (Aetheris743) [inline ARSIS-Unity/Assets/ARSIS/Core/EventManager/BasesArsisEvent.cs:1]: This is more of a question. While we are renaming the eventmanager, should we name it `ARSISEventSystem` or put it in a nested namespace like `ARSIS.Core.EventSystem`?

**Review by Aetheris743 — APPROVED** [review]
Looks good

File: ARSIS-Unity/Assets/ARSIS/Core/EventManager/BasesArsisEvent.cs
Comment (cburnell) [reply ARSIS-Unity/Assets/ARSIS/Core/EventManager/BasesArsisEvent.cs:1]: We should probably move to that but as its own PR. Just wanted to get this in so we can get the build working as quick as possible

File: ARSIS-Rover/teensy/src/main.cpp
Comment (cburnell) [reply ARSIS-Rover/teensy/src/main.cpp:1]: No its a good catch I am going to format it better

**Pull request merged**